packages/relay/src/lib/services/hbarLimitService/index.ts

```diff
@@ -358,7 +358,9 @@ export class HbarLimitService implements IHbarLimitService {
       try {
         return await this.getSpendingPlanByIPAddress(requestDetails);
       } catch (error) {
-        this.logger.warn(error, `Failed to get spending plan for IP address '${ipAddress}'`);
+        const spendingPlanId = (await this.ipAddressHbarSpendingPlanRepository.findByAddress(ipAddress, requestDetails))
```

Comment:
This would always be `undefined`, no point in doing this fetch.

Reply:
ah, yes you are right. i was trying to fix this quickly and overlooked

Reply:
Removed